blog/tidy-finance-dummy-data/index.qmd

```diff
@@ -0,0 +1,422 @@
+
+Note that we do not put any meaningful structure on the macro tables because they can be freely downloaded from the original sources - check out [Accessing and Managing Financial Data](../../r/accessing-and-managing-financial-data.qmd) to replace the dummy data with the actual data.
+
+```{r}
+macro_tables <- c(
+  "cpi_monthly", "factors_ff3_daily", 
+  "factors_ff3_monthly", "factors_ff5_monthly", 
+  "factors_q_monthly", "industries_ff_monthly", 
+  "macro_predictors"
+)
+
+for (table_name in macro_tables) {
+  data_original <- tbl(tidy_finance, table_name) |>
+    collect() |> 
+    drop_na()
+  
+  if ("month" %in% names(data_original)) {
+    time_series <- time_series_months
+    date_column <- "month"
+  } else if ("date" %in% names(data_original)) {
+    time_series <- time_series_days
+    date_column <- "date"
+  }
+  
+  relevant_columns <- data_original |> 
+    select(-contains(c("month", "date"))) |> 
+    names()
+  
+  commands <- unlist(
+    map(
+      relevant_columns, 
+      ~rlang::exprs(!!..1 := runif(n()))
+    )
+  )
+  
+  data_dummy <- tibble(
+    !!sym(date_column) := time_series
+    ) |> 
+    mutate(
+      !!!commands
+    )
+  
+  dbWriteTable(
+    tidy_finance_dummy, 
+    table_name,
+    data_dummy,
+    overwrite = TRUE
+  )
+}
+```
+
+## Create stock dummy data
+
+Let us move on the core data used throughout the book: stock and firm characteristics. We first generate a table with a cross section of stock identifiers with unique `permno` and `gvkey` values, as well as associated `exchcd`, `exchange`, `industry`, and `siccd` values. The generated data is based on the characteristics of stocks in the `crsp_monthly` table of the original database, ensuring that the generated stocks roughly reflect the distribution of industries and exchanges in the original data, but the identifiers and corresponding exchanges or industries do not reflect actual firms. Similarly, the `permno`-`gvkey` combinations are purely nonsensical and should not be used together with actual CRSP or Compustat data. 
```

Comment:
Typo correction: "Let us move on **to** the core data used throughout the book: stock and firm characteristics."

Consistency issue: We commonly use "cross-section" instead of "cross section"